runtime/README.md

```diff
@@ -1833,6 +1811,62 @@ The size of the (encrypted) context is always exactly 128 bytes,
 and we will use the type `AES_GCM_CONTEXT` to represent `u8[128]` below.
 
 
+### CM_AES_GCM_SPDM_ENCRYPT_INIT
+
+Derives the AES key and IV following the [SPDM 1.4](https://www.dmtf.org/dsp/dsp0274) and [Secured Messages using SPDM 1.1](https://www.dmtf.org/dsp/dsp0277) specifications.
+
+Currently only supports AES-256-GCM.
+
+Additional authenticated data (AAD) can only be passed during the `INIT` command, so is limited to the maximum cryptographic mailbox data size (4096 bytes).
+
+The CMK must have been created for **HMAC** usage.
+
+The CMK passed in should be the SPDM major secret CMK created for HMAC usage. The key and IV used for encryption shall follow the [SPDM 1.4](https://www.dmtf.org/dsp/dsp0274) section 12.7 derivation with `key_length` 256 and `iv_length` 96.
+
+```
+EncryptionKey = HKDF-Expand(major-secret, bin_str5, key_length);
+IV = HKDF-Expand(major-secret, bin_str6, iv_length);
+bin_str5 = BinConcat(key_length, Version, "key", null);
+bin_str6 = BinConcat(iv_length, Version, "iv", null);
+```
+
+The provided 64-bit message counter will be XOR'd with the IV to produce the message IV.
+
+Note that it is **critical** that the same CMK and counter never be used more than once when encrypting or decrypting in SPDM mode as doing could compromise the plaintext of the messages.
+
+For the update and final operations, there are not any SPDM-specific commands; `CM_AES_GCM_ENCRYPT_UPDATE` and `CM_AES_GCM_ENCRYPT_FINAL` should be used.
+
+Command Code: `0x434D_5345` ("CMSE")
+
+*Table: `CM_AES_GCM_SPDM_ENCRYPT_INIT` input arguments*
+| **Name**           | **Type**           | **Description**                   |
+| ------------------ | ------------------ | --------------------------------- |
+| chksum             | u32                |                                   |
+| spdm version       | u8                 | The value should be equal to the  |
+|                    |                    | byte representation of the SPDM   |
+|                    |                    | version, e.g., 0x13 = SPDM 1.3    |
+| counter endianness | u8                 | XOR the counter with the IV using |
```

Comment:
277 1.2 requires little endianness, so might be good to state that.

Reply:
Thanks. Let me know if what I wrote clarifies this.

Reply:
Yes, that should work.